client/src/api/block_builder/input_selection/core/requirement/ed25519.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{alias::is_alias_transition, InputSelection, Requirement};
+use crate::{
+    block::{address::Address, output::AliasTransition},
+    error::{Error, Result},
+    secret::types::InputSigningData,
+};
+
+impl InputSelection {
+    // Checks if a selected input unlocks a given ED25519 address.
+    fn selected_has_ed25519_address(&self, input: &InputSigningData, address: &Address) -> bool {
```

Comment:
Should this maybe be renamed to `selected_unlocks_ed25519_address()`?

Reply:
Done